### DISCUSSION TITLE
MagickImage ColormapSize "exception"

### DISCUSSION THREAD

**Discussion by Gounlaf** [dlemstra/Magick.NET#1533] [General] [answered]
:wave: 

Following this discussion https://github.com/dlemstra/Magick.Native/discussions/38, it appears that it's the only method that returns -1 when an error happens (whereas other has a exception).

Should it be _harmonized_ like others method?

Regards.

**Comment by dlemstra** [comment]
This discussion should probably by done in the Magick.Native but I can understand why you brought it here. I wonder why I chose `-1`. It probably makes more sense to return `0` there instead when the image does not have a color map. I don't think we need to throw an exception.

**Reply by Gounlaf** [reply]
> but I can understand why you brought it here

I like :ping_pong: 😅

**Reply by dlemstra** [reply, marked as answer]
I will probably need to include that change in the `14.0.0` changes of Magick.NET
